paper-dialog-scrollable.html

```diff
@@ -89,34 +116,22 @@
     attached: function() {
       this.classList.add('no-padding');
       // Set itself to the overlay sizing target
-      this.parentNode.sizingTarget = this.scrollTarget;
+      this.dialogElement.sizingTarget = this.scrollTarget;
       // If the host is sized, fit the scrollable area to the container. Otherwise let it be
       // its natural size.
-      // FIXME use resizeable
-      this.async(function() {
+      requestAnimationFrame(function() {
         if (this.offsetHeight > 0) {
           this.$.scrollable.classList.add('fit');
         }
         this._scroll();
-      }, 100);
+      }.bind(this));
     },
 
     _scroll: function() {
-      if (this.scrollTarget.scrollTop > 0) {
-        this.classList.add('is-scrolled');
-      } else {
-        this.classList.remove('is-scrolled');
-      }
-      if (this.scrollTarget.offsetHeight < this.scrollTarget.scrollHeight) {
-        this.classList.add('can-scroll');
-      } else {
-        this.classList.remove('can-scroll');
-      }
-      if (this.scrollTarget.scrollTop + this.scrollTarget.offsetHeight < this.scrollTarget.scrollHeight) {
-        this.classList.remove('scrolled-to-bottom');
-      } else {
-        this.classList.add('scrolled-to-bottom');
-      }
+      this.toggleClass('is-scrolled', this.scrollTarget.scrollTop > 0);
+      this.toggleClass('can-scroll', this.scrollTarget.offsetHeight < this.scrollTarget.scrollHeight);
+      this.toggleClass('scrolled-to-bottom',
+        this.scrollTarget.scrollTop + this.scrollTarget.offsetHeight >= this.scrollTarget.scrollHeight);
```

Comment:
Thanks, this method is a lot easier to read without all of the conditional statements :eyeglasses: